mozilla_django_oidc_db/backends.py

```diff
@@ -20,13 +22,15 @@ class OIDCAuthenticationBackend(SoloConfigMixin, _OIDCAuthenticationBackend):
     as unique identifier (default `sub`).
     """
 
-    def __init__(self, *args, **kwargs):
-        self.config = OpenIDConnectConfig.get_solo()
+    def __getattribute__(self, attr):
+        if attr.startswith("OIDC"):
+            return self.get_settings(attr, None)
+        return super().__getattribute__(attr)
 
-        if not self.config.enabled:
-            return
+    def __init__(self, *args, **kwargs):
+        self.UserModel = get_user_model()
 
-        super().__init__(*args, **kwargs)
+        return
```

Comment:
the `return` call at the end of `__init__` is not needed.

I'm also wondering if it's correct that the `super().__init__(*args, **kwargs)` is no longer called. If it's not needed, can you add a comment to document why it's not needed?

Reply:
Yes, the issue was that all the settings were initialized in the init of the parent: https://github.com/mozilla/mozilla-django-oidc/blob/master/mozilla_django_oidc/auth.py#L46, which would call the override of `get_settings`, which would call `get_solo`, etc. I'll add a comment explaining this